dotcom-rendering/src/web/components/Snap.tsx

```diff
@@ -14,6 +15,15 @@ export const Snap = ({ snapData }: Props) => {
 		return <></>;
 	}
 
+	// Emotion doesn't parse this conditional CSS correctly which is used by some thrashers to check
+	// if theres a pageskin.
+	if (snapData.embedCss !== undefined) {
+		snapData.embedCss = snapData.embedCss.replaceAll(
+			'body:not(.has-active-pageskin)',
+			'body:not(.has-active-pageskin) &',
```

Comment:
I presume this is a commercial classname? Do you think there's a way to break this coupling? It seems very brittle

Reply:
Ah, I think I see the use case now (sorry, it was clear before, I just didn't get it). 

So the problem here is that Thrashers are using a style of css that Emotion does not support. I think in this case it might be better to push back and make the request that this change happens at source, in the Thrasher code directly.

I think this is good for several reasons

- It educates those teams about the problem, making it and others like it less likely in future
- It is less code in DCR
- We want to encourage better links to the teams writing this code, this will help us improve Thrashers more
- and perhaps most importantly, it doesn't start a pattern where DCR mutates the css being passed to it to fix/enhance/add things. Those sorts of patterns have a habit of growing!

The downside to this is DCR will break when passed this code but I think that early failure is desirable, esp. now when we're not live and can get away with surfacing things like this using errors

Reply:
I would expect pageskins are the full screen takeovers we still occasionally see (though arguably we shouldn't) running on the site.

When this happens all the margins around the edge of the page are given to ad space, and the site feels like it's 'floating' in the middle of an ad - My guess is when this happens they probably don't want the protruding colours covering the ad!

If I find a screenshot I'll send it here for future reference

Reply:
> So the problem here is that Thrashers are using a style of css that Emotion does not support. I think in this case it might be better to push back and make the request that this change happens at source, in the Thrasher code directly.

I'm not quite sure its possible to fix it at source, atleast not while Frontend is still serving Fronts. Designers would have to add the `&` themselves which isn't valid CSS.

Reply:
> **Note**
> There’s two kinds of page skins:
> 1. the good ones, in the UK, where we know at server-render time that they will be around
> 2. the not-so-good ones, where anything goes and are entirely client-side, mainly in Australia

Reply:
Also, looking through some thrasher source code it seems clear that there are a number of expectations made around the html structure and what classnames might be available on the page. I can see code expecting `fc-container__inner`, `fc-container--thrasher`, etc. This code is not DCR safe.

This makes me think that we have two possible routes open to us:

1) We take the path that was taken for interactives on DCR, we amend the DCR code to try to accommodate the expectations from the html and css that we are trying to render on the page or
2) We ask the teams creating this content to edit their code to make it work on DCR

I think that our experience with option 1 to date tells us that this was not an ideal solution. Even so, in that case (interactive articles) we had such a long tail of content that this was the best option.

But there is a very limited number of thrashers so I think editing them is a viable option. I think this is the choice we should be making when we find errors with how they render on DCR.

Reply:
> Do you have more information on what the original error was that meant we need to make this change? I wasn't able to replicate it locally by removing this code

![3030 code ashleigh rocks_Front_url=https___www theguardian com_uk (3)](https://user-images.githubusercontent.com/21217225/174837183-963f038b-bae0-4d72-8066-90b43313be00.png)
With ampersans added ^

![3030 code ashleigh rocks_Front_url=https___www theguardian com_uk (2)](https://user-images.githubusercontent.com/21217225/174837275-5ef8a88e-0237-4c86-be0a-c4f6bb47d29c.png)
Without ampersans added ^

Without adding the ampersans Thrashers use the pageskin version of their styles and not the standard version.

> Also, looking through some thrasher source code it seems clear that there are a number of expectations made around the html structure and what classnames might be available on the page. I can see code expecting fc-container__inner, fc-container--thrasher, etc. This code is not DCR safe.

Theres no denying that there still is a lot of little visual bugs with a lot of thrashers from where they depend on styles that used to exist in Frontend

> If the extra ampersand isn't valid css I'm curious why DCR needs to add it to keep Emotion happy 🙃

Because its part of the emotion Syntax for doing nested selectors and not part of standard CSS syntax. https://emotion.sh/docs/nested 

> We ask the teams creating this content to edit their code to make it work on DCR

I agree with that, and thats probably how we'll go about resolving all of the other little issues with thrashers, but this one particular issue I don't think theres an easy way for us to fix them at the source